cmd/clusterctl/client/tree/discovery.go

```diff
@@ -69,10 +70,14 @@ func Discovery(ctx context.Context, c client.Client, namespace, name string, opt
 	}
 
 	// Adds control plane
-	controlPLane, err := external.Get(ctx, c, cluster.Spec.ControlPlaneRef, cluster.Namespace)
-	if err == nil {
-		tree.Add(cluster, controlPLane, ObjectMetaName("ControlPlane"), GroupingObject(true))
+	if cluster.Spec.ControlPlaneRef == nil {
+		return nil, errors.Errorf("controlplane ref cannot be nil")
 	}
+	controlPlane, err := external.Get(ctx, c, cluster.Spec.ControlPlaneRef, cluster.Namespace)
+	if err != nil {
+		return nil, err
```

Comment:
Following up on https://github.com/kubernetes-sigs/cluster-api/pull/4343#discussion_r613555044, this now returns error as well.

Reply:
Perhaps we could `errors.Wrap` this to add some extra information to make it easier to trace the error when it occurs?

Reply:
I thought about it, but since no other errors in this function were wrapped I kept this one as is. In other words if we wrap the error on this line, should I go ahead and wrap the other ones as well (e.g. line 85, 119, 124, etc)?